patches/react-native-webview+11.13.0.patch

```diff
@@ -0,0 +1,129 @@
+     }
+   }
+ 
++  @ReactProp(name = "androidMicrophoneAccessDisabled")
++  public void setMicrophoneAccessDisabled(WebView view, boolean disabled) {
++    ((RNCWebView) view).getCustomSettings().setMicrophoneAccessDisabled(disabled);
++  }
++
++  @ReactProp(name = "androidCameraAccessDisabled")
++  public void setCameraAccessDisabled(WebView view, boolean disabled) {
++    ((RNCWebView) view).getCustomSettings().setCameraAccessDisabled(disabled);
++  }
++
+   @ReactProp(name = "androidLayerType")
+   public void setLayerType(WebView view, String layerTypeString) {
+     int layerType = View.LAYER_TYPE_NONE;
+@@ -1229,6 +1239,10 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+ 
+         if (androidPermission != null) {
+           if (ContextCompat.checkSelfPermission(mReactContext, androidPermission) == PackageManager.PERMISSION_GRANTED) {
++              if ((androidPermission.equals(Manifest.permission.RECORD_AUDIO) && ((RNCWebView) mWebView).getCustomSettings().getMicrophoneAccessDisabled()) ||
++                (androidPermission.equals(Manifest.permission.CAMERA) && ((RNCWebView) mWebView).getCustomSettings().getCameraAccessDisabled())) {
++                continue;
++            }
+             grantedPermissions.add(requestedResource);
+           } else {
+             requestedAndroidPermissions.add(androidPermission);
+@@ -1256,7 +1270,6 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+ 
+       if (ContextCompat.checkSelfPermission(mReactContext, Manifest.permission.ACCESS_FINE_LOCATION)
+         != PackageManager.PERMISSION_GRANTED) {
+-
+         /*
+          * Keep the trace of callback and origin for the async permission request
+          */
+@@ -1447,6 +1460,17 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+     protected boolean nestedScrollEnabled = false;
+     protected ProgressChangedFilter progressChangedFilter;
+ 
++    /**
++       *  It is impossibile to override WebView getSettings(), so RNCWebView has getCustomSettings()
++       */
++    private CustomWebSettings customWebSettings;
++    public CustomWebSettings getCustomSettings() {
++      if (customWebSettings == null) {
++        customWebSettings = new CustomWebSettings();
++      }
++      return customWebSettings;
++    }
++
+     /**
+      * WebView must be created with an context of the current activity
+      * <p>
+@@ -1735,5 +1759,26 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+         return waitingForCommandLoadUrl;
+       }
+     }
++
++    private class CustomWebSettings {
++      private boolean microphoneAccess = true;
++      private boolean cameraAccess = true;
++
++      private boolean getMicrophoneAccessDisabled() {
++        return !microphoneAccess;
++      }
++
++      private boolean getCameraAccessDisabled() {
++        return !cameraAccess;
++      }
++
++      private void setMicrophoneAccessDisabled(boolean b) {
```

Comment:
It seems like the names are flipped (setMicrophone mutates camera and the other way round at line 80).

Reply:
good catch!
fixed a39d8cec678e454e8931b6c14cf65b60ebc85e3c